### PR TITLE
Adding admin service and client script

### PR DESCRIPTION
**Pull request by lmanohara**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag [hunk 1]
```diff
@@ -76,6 +76,50 @@ var getVersionHashId = function(applicationName, versionName, tenantDomain){
...
+    }
+    return appCountObj;
+};
+
+var updateContainerSpecification = function (tenantDomain, applicationName, applicationRevision, memory, cpu) {
+    try {
+        var versionHashId = getVersionHashId(applicationName, applicationRevision, tenantDomain);
+        ApplicationManager.updateContainerSpecification(versionHashId, parseInt(memory), parseInt(cpu));
+    } catch (e) {
+        throw e;
```
Comment (manjulaRathnayaka) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: if we are not logging the error, why we have to catch it again?

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 2]
```diff
@@ -1586,4 +1588,36 @@ public void whiteListTenant(Connection dbConnection, int tenantId, int maxAppCou
...
+            throws AppCloudException {
+        PreparedStatement preparedStatement = null;
+        boolean updated = false;
+
+        try {
+            preparedStatement = dbConnection.prepareStatement(SQLQueryConstants.UPDATE_APP_VERSION_CON_SPEC);
+            preparedStatement.setInt(1, memory);
+            preparedStatement.setInt(2, cpu);
+            preparedStatement.setString(3, versionHashId);
+            updated = preparedStatement.execute();
```
Comment (hisamith) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: just return here. No need to define a variable and return it later

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag [hunk 3]
```diff
@@ -76,6 +76,50 @@ var getVersionHashId = function(applicationName, versionName, tenantDomain){
...
+            versionObj.versionName = versions.get(i).versionName;
+            versionObj.status = versions.get(i).status;
+            versionObj.conSpecCpu = versions.get(i).conSpecCpu;
+            versionObj.conSpecMemory = versions.get(i).conSpecMemory;
+            versionObj.isWhiteListed = versions.get(i).isWhiteListed;
+
+            applicationVersions.push(versionObj);
+        }
+    } catch (e) {
+        throw e;
```
Comment (hisamith) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: Why are you catching the exception and throw the same exception here??

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag [hunk 4]
```diff
@@ -76,6 +76,50 @@ var getVersionHashId = function(applicationName, versionName, tenantDomain){
...
+    return applicationVersions;
+};
+
+var getTenantAppMaxCount = function (tenantDomain) {
+    var tenantId = modManager.getTenantIdForGivenTenantDomain(tenantDomain);
+    var appCountObj = {};
+    try {
+        appCountObj.maxAppCount = ApplicationManager.getMaxAppCountForWhiteListedTenants(tenantId);
+    } catch (e) {
+        throw e;
```
Comment (hisamith) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: Why are you catching the exception and throw the same exception here??

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 1 above)
Comment (hisamith) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: Why are you catching the exception and throw the same exception here??

File: modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag [hunk 5]
```diff
@@ -65,12 +61,47 @@ var log = new Log('site/blocks/admin/admin.jag');
...
+        }
+    } else if (action == "getApplicationVersions") {
+        var tenantDomain = request.getParameter("tenantDomain");
+        var applicationName = request.getParameter("applicationName");
+
+        try {
+            var applicationVersionList = adminModule.getApplicationVersions(tenantDomain, applicationName);
+            print(applicationVersionList);
+        } catch (e) {
+            response.status = 400;
```
Comment (hisamith) [inline modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag]: What is happening to the actual error message? Is it logged?

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java (hunk 2 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Fixed in following pull request.
https://github.com/wso2/app-cloud/pull/417

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 3 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: I fixed this issue, logged and throw the exception to catch from the block layer.

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 4 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: I fixed this issue, logged and throw the exception to catch from the block layer.

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 1 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: I fixed this issue, logged and throw the exception to catch from the block layer.

File: modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag (hunk 5 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag]: This is logged in module layer.

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 3 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: I fixed this issue, logged and throw the exception to catch from the block layer.

File: modules/jaggeryapps/appmgt/src/modules/admin/admin.jag (hunk 4 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/admin/admin.jag]: I fixed this issue, logged and throw the exception to catch from the block layer.

File: modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag (hunk 5 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/blocks/admin/admin.jag]: This error is logged in module layer.

**Pull request merged**